File: 6-kyu/primorial-of-a-number-/python/solution.py
def is_prime(x):
    if x < 2:
        return False
    elif x == 2:
        return True
    elif x % 2 == 0:
        return False
    for j in range(3, x + 1, 2):
        if x % j == 0 and x != j:
            return False
    return True

def num_primorial(n):
    prime_count = 1
    ret = 1
    i = 0
    while prime_count <= n:
        if is_prime(i):
            ret *= i
            prime_count += 1
        i += 1
    return ret

File: 6-kyu/primorial-of-a-number-/swift/solution.swift
func isPrime(_ n: UInt) -> Bool {
    guard n > 1 else { return false }
    guard n != 2 else { return true }
    guard n % 2 == 1 else { return false }
    for i in stride(from: 3, through: n, by: 2) {
        if n % i == 0 && n != i {
            return false
        }
    }
    return true
}

func numPrimoral(_ number: UInt) -> UInt {
    var primeCount: UInt = 1, ret: UInt = 1, i: UInt = 0
    while primeCount <= number {
        if isPrime(i) {
            ret *= i
            primeCount += 1
        }
        i += 1
    }
    return ret
}